File: LeetCode/118-pascals-triangle/pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> sol {{1}};
        for (int i = 1; i < numRows; i++) {
            sol.push_back(gen(sol[i - 1]));
        } //for
        return sol;
    } //gen

    vector<int> gen(vector<int>& prev) {
        vector<int> curr (1,1);
        for (int i = 0; i < prev.size() - 1; i++) {
            curr.push_back(prev[i] + prev[i + 1]);
        } //for
        curr.push_back(1);
        return curr;
    } //gen
};

File: LeetCode/118-pascals-triangle/pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
       List<List<Integer>> sol = new ArrayList <> ();
       sol.add(new ArrayList <Integer> (Arrays.asList(1)));
       
       for (int i = 1; i < numRows; i++) {
           List<Integer> curr = new ArrayList <> ();
           curr.add(1);
           for (int j = 0; j < sol.get(i - 1).size() - 1; j++) {
               curr.add(sol.get(i - 1).get(j) + sol.get(i - 1).get(j + 1));
           } //for
           curr.add(1);
           sol.add(curr);
       } //for

       return sol;
    } //gen
} //Sol